snapcraft/plugins/go.py

```diff
@@ -93,7 +93,8 @@ def get_pull_properties(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
-        self.build_packages.append("golang-go")
+        if "go" not in self.build_snaps:
```

Comment:
You might want to account for channels too, `go/<channel>` is the syntax, so maybe something like this would be better:
```python
if "go" not in self.build_snaps and not any((s.startswith("go/") for s in self.build_snaps)):
    self.build_packages.append("golang-go")
```

I will take care of bases and such in a future branch.

Reply:
Ok, I updated the code to support "go/foo" and "go=foo" in the go plugin.